metamorph/server.go

```diff
@@ -401,6 +401,22 @@ func (s *Server) putTransactionInit(ctx context.Context, req *metamorph_api.Tran
 	}
 
 	if rtr.BlockHash != nil {
+		// In case a transaction is submited to network outside of ARC and mined and now
+		// submitting the same transaction through arc endpoint we have problem.
+		// The transaction doesn't exist in metamorph so the call after this line
+		// (updating tx status to MINED) will fail as there is no transaction to update.
+		// In that case we take the transaction and store it first in metamorph database.
+		if err := s.store.Set(ctx, hash[:], &store.StoreData{
+			Hash:          &hash,
+			Status:        status,
+			CallbackUrl:   req.CallbackUrl,
+			CallbackToken: req.CallbackToken,
+			MerkleProof:   req.MerkleProof,
+			RawTx:         req.RawTx,
+		}); err != nil {
+			s.logger.Warn("Transaction already exists in db", slog.String("warn", err.Error()))
```

Comment:
1. Isn't it suposed to be ".Error" instead of Warn?
2. If the store is dynamodb the errors in store.Set can be because of `MarshalMap` and `dynamoCli.PutItem` so not sure the message is correct. 
The error might be not because of existing record.

Reply:
If we have the transaction already there it will return error but it's not really error for us. 

I'll check if there is anything to be cautious in dynamodb implmenentation.

Reply:
I guess the error may be something else than transaction already existing, not?

I'd put there something like this: `s.logger.Warn("Failed to store transaction", slog.String("hash", hash.String()), slog.String("warn", err.Error()))`

Reply:
yes but it's still error and not warn. Because you are handling error not warning.

Reply:
Ok, then rather this I guess
```go
s.logger.Error("Failed to store transaction", slog.String("hash", hash.String()), slog.String("err", err.Error()))
```